picby_graphql_api/src/modules/user/register/isEmailAlreadyExists.ts

```diff
@@ -0,0 +1,25 @@
+import { registerDecorator, ValidationOptions, ValidatorConstraint, ValidatorConstraintInterface } from 'class-validator';
+import { User } from '../../../entity/User';
+
+@ValidatorConstraint({ async: true })
+export class IsEmailAlreadyExistConstraint
+  implements ValidatorConstraintInterface {
+  validate(email: string) {
+    return User.findOne({where:{email}}).then(user => {
+      if (user) return false;
+      return true;
+    });
+  }
+}
+
+export function IsEmailAlreadyExist(validationOptions?: ValidationOptions) {
```

Comment:
name, could be e.g.:
DoesEmailExist / IsEmailInUse

also in the file and class name